libs/plugins/function.html_select_date.php

```diff
@@ -306,11 +306,36 @@ function smarty_function_html_select_date($params, Smarty_Internal_Template $tem
             $_html_months .= '<option value="">' . (isset($month_empty) ? $month_empty : $all_empty) . '</option>' .
                              $option_separator;
         }
+
+        $formatter = null;
+        $format_compare = '%m';
+        if (class_exists('IntlDateFormatter')) {
```

Comment:
Since `IntlDateFormatter` isn't included by default and the `strftime` deprecation notice is only a notice, wouldn't it be better to keep it as it was? We're adding complexity here, only fixing the deprecation notices on certain environments. I'm hoping PHP9 will include a way to handle this smoothly that is included by default?

Reply:
Unfortunately, I don't think that PHP 9.0 will handle it smoothly, from [https://wiki.php.net/rfc/deprecations_php_8_1](https://wiki.php.net/rfc/deprecations_php_8_1):
>The strftime() and gmstrftime() functions exhibit similar issues as strptime(), in that the formats they support, as well as their behavior, is platform-dependent. Unlike strptime(), these functions are available on Windows, though with a different feature set than on Linux. Musl-based distributions like Alpine do not support timezone-related format specifiers correctly. These functions are also locale-based, and as such may exhibit thread-safety issues.

> Once again date() or DateTime::format() provide portable alternatives, and IntlDateFormatter::format() provides a more sophisticated, localization-aware alternative.

> The proposal is to deprecate strftime() and gmstrftime() in favor of these alternatives.

So I guess this is the way we'll have to do it in PHP 9.0 and in the meantime, we can have `strftime` as a fallback. WDYT?

Reply:
Let me put it in another way: your change will still fail in PHP9, because `strftime` will be deleted by then (unless they reverse the decision). So what is the point of making the change in the first place?
Also: I know `strftime` is used a LOT in PHP libs, so the deprecation will trigger the need for a suitable alternative to be included in PHP9 by default. If that turns out to be `IntlDateFormatter`, we'll switch then.

Reply:
I agree with @atomiix , having `strftime` as fallback is a good idea. I'm not in favor of having deprecation notices only because it's acceptable. When you're working you don't want to be spamed by useless messages you can avoid.
Adding `IntlDateFormatter` as alternative is a good idea, but the real check should be `if (extension_loaded('intl')) {` for a better understanding.

Reply:
I also think avoiding any `strftime` would be beneficial. The intl extension is quite common nowadays, and Symfony has a polyfill (`symfony/polyfill-intl-icu`) which is an easy alternative if the intl extension is missing and provides a way of avoiding deprecation notices even without the extension.